File: MiniProtobufClient/SigninProcessor.cpp
#include "SigninProcessor.h"

using namespace BaseCmd;
SigninProcessor::SigninProcessor()
{
}

SigninProcessor::~SigninProcessor()
{
}

void SigninProcessor::Process(std::shared_ptr<MessageRoute> spRoute, std::shared_ptr<google::protobuf::Message> spMsg)
{
	rspSignin* p_rsp = reinterpret_cast<rspSignin*>(spMsg.get());

	std::string name = p_rsp->name();
	bool succ = p_rsp->succ();
	//std::cout << "signin rsp:" << name << " Succ:" << succ <<  std::endl;
}

File: AndroidProj/app/src/main/java/com/dd/androidprotobuf/SigninProcessor.java
package com.dd.androidprotobuf;

import android.content.Intent;

import com.google.protobuf.GeneratedMessageLite;

import BaseCmd.Cmd;

public class SigninProcessor extends  BaseProcessor {

    @Override
    void Process(MessageRoute route, GeneratedMessageLite msg)
    {
        Cmd.rspSignin rsp = (Cmd.rspSignin)msg;
        rsp.getName();
        rsp.getSucc();


        Intent i = new Intent(BaseProcessor.MessageFilter);
        i.putExtra(Intent.EXTRA_TEXT,rsp.getName() + " " + rsp.getSucc());

        //i.setData()
        MainActivity.Instance.sendBroadcast(i);
    }
}
